docs/content/sips/008-logging.md

```diff
@@ -0,0 +1,104 @@
+title = "SIP 008 - Runtime Logging"
+template = "main"
+date = "2022-12-09T11:36:36Z"
+---
+
+Summary: Unified runtime logging
+
+Owner: mail@etehtsea.me
+
+Created: December 09, 2022
+
+## Background
+
+## Current experience
+
+Currently, by default spin prints into `stdout` only the basic info:
+
+```
+$ spin up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  hello: http://127.0.0.1:3000/hello
+    A simple component that returns hello.
+```
+
+Application log is implicitly written into `~/.spin/<spin_app>/logs/<component_id>_std{err|out}.log`.
+The log for the spin itself has log level `ERROR` and outputs into `stderr`.
+
+To enable a full logging experience, a developer needs to provide `--follow-all` to duplicate app log into `stderr` and also `RUST_LOG=spin=<log_level>` to see the spin log in the `stderr`.
+
+
+## Proposal
+
+- Set default log level to `INFO`.
+- Set default log destination to `stdout/stderr`.
+- Provide a way for spin apps for logging (besides using `println!`/`eprintln!`).
```

Comment:
Yep, I think it would be handy to log using tracing from the guest modules in the consistent manner.

When I was playing with an example app on the fermyon cloud, sometimes it was hard to distinguish one log record from another as there are no timestamps or ids.

Though, `println!` (or any other "print to std") behaviour should remain unaffected.

Reply:
> When I was playing with an example app on the fermyon cloud, sometimes it was hard to distinguish one log record from another as there are no timestamps or ids.

This is mostly just a UI issue in Fermyon Cloud; we are in fact _recording_ more info for logging, just not displaying it yet. The trick with timestamps specifically is that some apps are already logging with timestamps; we'll probably need to apply some "timestamp dedupe" before unconditionally showing timestamps in the UI.